### PR TITLE
feat(api-client): request creation from sidebar

### PR DESCRIPTION
**Pull request by antlio**
this pr adds request creation button from the sidebar collection item as seen below:

image

along:
- updating add icon
- updating chevron right icon

in order to fix icon size and thickness inconsistencies below:

⊢ before / after
<div>
image
image
</div>

### CODE REVIEW

**Comment by cameronrohani** [comment]
awesome work! unrelated but but the watch mode icon feel slightly too thick - does it match the thickness of chevrons + icons on the left?

File: packages/components/src/components/ScalarIcon/icons/ChevronRight.svg
Comment (hwkr) [inline packages/components/src/components/ScalarIcon/icons/ChevronRight.svg:1]: If we're going to change the Chevron Right icon we also need to update the others so they match and then make sure all the chevrons look okay in their new placements. It looks like left already matches but up and down look bigger.

Google Chrome-2024-11-26-13-41-39@2x


```
# Query: Chevron\\w+
# Flags: CaseSensitive RegExp
# ContextLines: 1

48 results - 31 files

packages/api-client/src/components/CommandPalette/CommandPaletteExample.vue:
  84                class="text-c-3"
  85:               icon="ChevronDown"
  86                size="xs" />

packages/api-client/src/components/CommandPalette/CommandPaletteRequest.vue:
  154                class="text-c-3"
  155:               icon="ChevronDown"
  156                size="xs" />

  170                class="text-c-3"
  171:               icon="ChevronDown"
  172                size="xs" />

packages/api-client/src/components/CommandPalette/CommandPaletteServer.vue:
  90              class="text-c-3"
  91:             icon="ChevronDown"
  92              size="xs" />

packages/api-client/src/components/CommandPalette/CommandPaletteTag.vue:
  75              class="text-c-3"
  76:             icon="ChevronDown"
  77              size="xs" />

packages/api-client/src/components/CommandPalette/TheCommandPalette.vue:
  321            <ScalarIcon
  322:             icon="ChevronLeft"
  323              size="md"

packages/api-client/src/components/DataTable/DataTableInputSelect.vue:
  91            <ScalarIcon
  92:             icon="ChevronDown"
  93              size="xs" />

packages/api-client/src/components/EnvironmentSelector/EnvironmentSelector.vue:
  48              class="size-3"
  49:             icon="ChevronDown"
  50              thickness="3" />

packages/api-client/src/components/ImportCollection/WorkspaceSelector.vue:
  71              class="size-3"
  72:             icon="ChevronDown"
  73              thickness="3" />

packages/api-client/src/components/ViewLayout/ViewLayoutCollapse.vue:
  26            class="text-c-3 group-hover:text-c-1 group-focus-visible:outline ui-open:rotate-90 ui-not-open:rotate-0 rounded-[1px] outline-offset-2"
  27:           icon="ChevronRight"
  28            size="sm"

packages/api-client/src/views/Cookies/Cookies.vue:
  117                    }"
  118:                   icon="ChevronRight"
  119                    size="sm"

  137                        }"
  138:                       icon="ChevronRight"
  139                        size="sm"

packages/api-client/src/views/Request/RequestSidebar.vue:
  252                    class="text-c-3 hidden text-sm group-hover:block"
  253:                   icon="ChevronRight"
  254                    size="sm"

packages/api-client/src/views/Request/RequestSidebarItem.vue:
  372                  class="text-c-3 text-sm"
  373:                 icon="ChevronRight"
  374                  size="sm"

packages/api-client/src/views/Request/components/WorkspaceDropdown.vue:
  104                class="size-3"
  105:               icon="ChevronDown"
  106                thickness="3" />

packages/api-client/src/views/Request/RequestSection/RequestBody.vue:
  424                <ScalarIcon
  425:                 icon="ChevronDown"
  426                  size="xs"

packages/api-client/src/views/Request/RequestSection/RequestAuth/OAuthScopesInput.vue:
  66              class="text-c-3 group-hover/scopes-accordion:text-c-2"
  67:             :icon="open ? 'ChevronDown' : 'ChevronRight'"
  68              size="xs" />

packages/api-client/src/views/Request/RequestSection/RequestAuth/RequestAuth.vue:
  292                    class="min-w-3 ml-auto mr-2.5"
  293:                   icon="ChevronDown"
  294                    size="xs" />

packages/api-reference/src/components/ShowMoreButton.vue:
  20        class="show-more-icon"
  21:       icon="ChevronDown" />
  22    </button>

packages/api-reference/src/components/Section/CompactSection.vue:
  38        <ScalarIcon
  39:         :icon="open ? 'ChevronDown' : 'ChevronRight'"
  40          size="sm" />

packages/api-reference/src/components/Section/SectionAccordion.vue:
  40            class="section-accordion-chevron"
  41:           :icon="open ? 'ChevronDown' : 'ChevronRight'" />
  42        </DisclosureButton>

packages/api-reference/src/components/Section/SectionContainerAccordion.vue:
  14            class="section-accordion-chevron"
  15:           :icon="open ? 'ChevronDown' : 'ChevronRight'" />
  16          <div class="section-accordion-title">

packages/api-reference/src/components/Sidebar/SidebarElement.vue:
  104            class="toggle-nested-icon"
  105:           :icon="open ? 'ChevronDown' : 'ChevronRight'"
  106            :label="`${open ? 'Collapse' : 'Expand'} ${item.title}`"

packages/api-reference/src/features/BaseUrl/ServerUrlSelect.vue:
  54          v-if="options.length > 1"
  55:         icon="ChevronDown"
  56          size="xs" />

packages/api-reference/src/features/BaseUrl/ServerVariablesSelect.vue:
  46        <ScalarIcon
  47:         icon="ChevronDown"
  48          size="xs" />

packages/api-reference/src/features/Operation/components/ParameterListItem.vue:
  51            class="parameter-item-icon"
  52:           :icon="open ? 'ChevronDown' : 'ChevronRight'"
  53            size="md"

packages/api-reference/src/legacy/components/SecuritySchemeScopes.vue:
  51            <ScalarIcon
  52:             :icon="open ? 'ChevronUp' : 'ChevronDown'"
  53              size="sm" />

packages/api-reference/src/legacy/components/SecuritySchemeSelector.vue:
  165          <ScalarIcon
  166:           icon="ChevronDown"
  167            size="xs" />

packages/components/src/components/ScalarIcon/icons/index.ts:
   9    'Checkmark',
  10:   'ChevronDown',
  11:   'ChevronLeft',
  12:   'ChevronRight',
  13:   'ChevronUp',
  14    'Clipboard',

packages/components/src/components/ScalarListbox/ScalarListbox.stories.ts:
  47          </span>
  48:         <ScalarIcon icon="ChevronDown" size="sm" class="ml-1 ui-open:rotate-180" />
  49        </div>

  83          </span>
  84:         <ScalarIcon icon="ChevronDown" size="sm" class="ml-1 ui-open:rotate-180" />
  85        </div>

packages/components/src/components/ScalarMenu/ScalarMenuButton.vue:
  19        class="text-c-3 group-hover/button:text-c-1"
  20:       :icon="open ? 'ChevronUp' : 'ChevronDown'"
  21        size="sm"

packages/components/src/components/ScalarMenu/ScalarMenuTeamPicker.vue:
  37            class="ml-auto text-c-2"
  38:           icon="ChevronDown"
  39            size="xs"

packages/use-codemirror/src/hooks/useCodeMirror.ts:
  394            const vnode = h(ScalarIcon, {
  395:             icon: open ? 'ChevronDown' : 'ChevronRight',
  396              size: 'xs',

  414              const vnode = h(ScalarIcon, {
  415:               icon: open ? 'ChevronDown' : 'ChevronRight',
  416                size: 'xs',

```

**Review by hwkr — CHANGES_REQUESTED** [review]
Just need to update the other Chevrons I think

File: packages/components/src/components/ScalarIcon/icons/ChevronRight.svg
Comment (antlio) [reply packages/components/src/components/ScalarIcon/icons/ChevronRight.svg:1]: up and down updated!

**Review by hwkr — APPROVED** [review]
Looks great! Thank you for updating those :)

**Review by hanspagel — APPROVED** [review]
n1c3

**Pull request merged**